arch/arm/Makefile

```diff
@@ -28,6 +28,10 @@ else
 arch-y += -D__LINUX_ARM_ARCH__=$(CONFIG_SYS_ARM_ARCH)
 endif
 
+ifeq ($(CONFIG_XEN),y)
+arch-y += -D__XEN__
```

Comment:
__XEN__ is meant to be used by the hypervisor only. It protects anything that is not meant to be stable from a domain PoV and will default the ABI to the latest one. This will have an impact longer term as you update the headers. As U-boot is running in a domain, he needs to use the stable ABI.


As U-boot will run in a domain, it should only used the stable ABI. AFAICT, you are

Reply:
Yes, I understand that both __XEN__ and __XEN_TOOLS__ are there for good purpose. But to me it had cons and pros of either hardcoding the values or using __XEN__ to get those from the relevant header

Reply:
I think you missed my point. By using anything guarded by those define, you effectively tie U-boot to a specific version of Xen. To give a concrete example, if tomorrow we decide to rework the memory layout), then U-boot is going to break.

How do you plan to deal with that in upstream U-boot?

Reply:
I understand that and it seems I didn't miss the point. Yes, we do tie to this specific version of Xen by copying its headers and using __XEN__. But what else can we do? Define those constants WRT, again, to the version of Xen we are running on. So, both ways will require changing either Xen header(s) to be up to date with Xen or changing the relevant constant in `include/configs/xenguest_arm64.h`

Reply:
Well, when you want to do the right thing from scratch, it will always take some effort...
I can see another alternative. I am not happy with it but I will mention it for fairness.
You could require the user to provide which version of Xen he/she is using. This would be used to define the list of RAM banks used.